### DISCUSSION TITLE
Unknown profiling option -fprofile-filter-files

### DISCUSSION THREAD

**Discussion by m-wachter** [ccache/ccache#1284] [Questions & answers] [answered]
Hello,

I want to use ccache version 4.8 in an existing build-system witch calls gcc with the compiler option -fprofile-filter-files.
Ccache seems always be calling the real compiler if this option is set:
```
[2023-05-09T19:27:08.807834 1386680] Compiler: /opt/sysroots/x86_64-tdxsdk-linux/usr/bin/i686-tdx-linux/i686-tdx-linux-g++
[2023-05-09T19:27:08.807835 1386680] Compiler type: gcc
[2023-05-09T19:27:08.807883 1386680] Unknown profiling option: -fprofile-filter-files=src/.*
[2023-05-09T19:27:08.807919 1386680] Failed; falling back to running the real compiler
```

I tried to ignore this option by setting the `CCACHE_IGNOREOPTIONS` environment variable. 
```
[2023-05-09T19:27:08.807737 1386680] Config: (environment) ignore_options = -fprofile-filter-files=*
```
but it seems to have no effect.

How can I ignore the -fprofile-filter-files option correctly?

Thanks in advance.

Michael.

**Comment by jrosdahl** [comment, marked as answer]
> How can I ignore the -fprofile-filter-files option correctly?

The only way you can do that currently is by adding [`--ccache-skip`](https://ccache.dev/manual/4.8.html#_extra_options) before the option.

> I tried to ignore this option by setting the `CCACHE_IGNOREOPTIONS` environment variable.

`CCACHE_IGNOREOPTIONS` is only about excluding options from the input hash; it does not affect other processing of options.

However, I think that it would make sense to improve `CCACHE_IGNOREOPTIONS` so that matching options are not processed at all, similar to `--ccache-skip`.

**Reply by m-wachter** [reply]
Hello Joel,

Using `--ccache-skip` works fine. 

Thank thank you for the fast answer.

Michael.

**Reply by jrosdahl** [reply]
> However, I think that it would make sense to improve CCACHE_IGNOREOPTIONS so that matching options are not processed at all, similar to --ccache-skip.

Done in 2737d79e282de72d1b29128f437173e4892ced5e.
